hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -650,6 +665,10 @@ private boolean checkIfJoinRequestFromAnExistingMember(JoinMessage joinMessage,
                 }
 
                 // send members update back to node trying to join again...
+                boolean deferPartitionProcessing = false;
+                if (isMemberRestartingWithPersistence(member)) {
+                    deferPartitionProcessing = true;
+                }
```

Comment:
Not really a thing, but statement can be simplified maybe
```suggestion
                boolean deferPartitionProcessing = isMemberRestartingWithPersistence(member);
```